webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -132,56 +254,123 @@ trait PromiseGenerationService extends Loggable {
 
     import HooksImplicits._
 
+
+    def buildFullyOneNode(
+        nodeId            : NodeId
+      , nodeConfigId      : NodeConfigId
+      , rootNodeId        : NodeId
+      , nodeConfigs       : Map[NodeId, NodeConfiguration]
+      , allLicenses       : Map[NodeId, CfeEnterpriseLicense]
+      , techniqueResources: TechniqueResources
+      , globalPolicyMode  : GlobalPolicyMode
+      , generationTime    : DateTime
+      , expectedReports   : NodeExpectedReports
+      , remainingNodes    : AtomicInt
+    ): Box[NodeExpectedReports] = {
+
+    val traceId = "[" + nodeId.value + "]"
+
+      val writeTime           =  System.currentTimeMillis
+      for {
+        writtenNodeConfigs    <- writeNodeConfigurations(rootNodeId, Map(nodeId -> nodeConfigId), nodeConfigs, allLicenses, techniqueResources, globalPolicyMode, generationTime) ?~!"Cannot write nodes configuration for node '${nodeId.value}'"
+        timeWriteNodeConfig   =  (System.currentTimeMillis - writeTime)
+        _                     =  PolicyLogger.debug(s"${traceId} Node configuration for node '${nodeId.value}' written in ${timeWriteNodeConfig} ms.")
+
+        saveExpectedTime      =  System.currentTimeMillis
+        savedExpectedReports  <- saveOneExpectedReports(expectedReports) ?~! "Error when saving expected reports for node '${nodeId.value}'"
+        endTime               =  System.currentTimeMillis
+        timeSaveExpected      =  endTime - saveExpectedTime
+        _                     =  PolicyLogger.debug(s"${traceId} Node expected reports for node '${nodeId.value}' saved in base in ${timeSaveExpected} ms.")
+        r                     =  remainingNodes.decrementAndGet()
+        _                     =  PolicyLogger.info(s"Configuration for node ${nodeId.value} updated in ${endTime - writeTime} ms (remains ${r} nodes to update)")
+      } yield {
+        savedExpectedReports
+      }
+    }
+
+    val maxParallelismEnv = "rudder.generation.maxParallelism"
```

Comment:
environment variable is certainly not what we want: we want something that can be change at runtime easily.